### DISCUSSION TITLE
geotagged images onto map?

### DISCUSSION THREAD

**Discussion by hardreddata** [TerriaJS/terriajs#5075] [Ideas] [answered]
I am wondering about doing something like https://communityhealthmaps.nlm.nih.gov/tag/raster-image-marker/ in Terria

I am a bit surprised I haven't found anything like a _geotagged photo wms maker_ anywhere. 

The idea gets a mention in here https://www.isprs.org/proceedings/2011/gi4dm/pdf/OP24.pdf but this was a while ago.

It looks like https://www.digikam.org/ has some smarts.

Maybe the move is to have end users drop their photos in a folder and I poll it to maintain the vector layer and the symbology (here photos) in GeoServer? Or I could cook up something like the ABC photo stories layer.

Drag & drop to the map would be special but probably outside my grasp.

Any advice is welcome.

Thanks!

**Comment by tephenavies** [comment, marked as answer]
That sounds like a cool feature. Geotag information is stored in EXIF metadata so some playing with maybe https://www.npmjs.com/package/exif-parser might give you some results. I think drag & drop a geotagged photo to show a point and have the photo in feature info should be fairly easy to implement. Let me know if you'd like some pointers.

**Comment by hardreddata** [comment]
Thanks.

Fun fact: the Outlook phone app will strip all your exif tags when you email your photos!

**Comment by DavidHine** [comment]
Hi Russell and Steve
We have geotagged images in a Leaflet map at https://geoportal.landandwater.com.au It is a demo.
login keys are user and user (on that page). The default view is Malaysia but we put in some from Bundaberg region as well.
There is a plugin for geotagged images. We use Lychee (https://lychee.electerious.com/) as the image library but others will work. The development was done by a contractor. I don't recall all the details of configuration, but think that the documentation at Leaflet will probably provide what is needed.
Then there is the integration with Terriajs .... but I am guessing that would not be complex.
